tooling/minimize/src/rvalue.rs

```diff
@@ -413,14 +413,19 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                         let root = GcCow::new(expr);
                         PlaceExpr::Index { root, index: i }
                     }
+                    stable_mir::mir::ProjectionElem::ConstantIndex {
+                        offset,
+                        from_end: false,
+                        ..
```

Comment:
Please don't ignore `min_length`, that seems dangerous.

We should raise UB if the length is smaller than `min_length`.

Reply:
Makes sense, but this is actually rather a convoluted change, so I'd first like to know the purpose & be sure it is dangerous. Because we do generate UB on invalid indices anyway, so this just seems like additional UB? Or is there a case where we want UB, but it is only encoded by the `min_length`, but the index is valid? For arrays it is always exactly the size.
And the implementation is convoluted, since this now means, the folding cannot just output a `PlaceExpr`, there needs to be code generated to raise UB, which must include code to get the length of the indexable, which then depends on whether it is an array or slice.

Reply:
I am honestly not entirely sure what the point of the `min_length` field is, but Miri uses this for a dedicated UB check, and that seems better than just ignoring it. I wrote https://github.com/rust-lang/rust/pull/131285 to clarify this in the rustc docs.

> which must include code to get the length of the indexable, which then depends on whether it is an array or slice.

You can share that code with `Rvalue::Len`.